apps/meteor/ee/server/models/raw/LivechatRooms.ts

```diff
@@ -293,4 +318,417 @@ export class LivechatRoomsRawEE extends LivechatRoomsRaw implements ILivechatRoo
+						departmentId: {
+							$exists: true,
+						},
+						ts: {
+							$lt: end,
+							$gte: start,
+						},
+						...extraQuery,
+					},
+				},
+				{
+					$group: {
+						_id: '$departmentId',
+						total: { $sum: 1 },
+					},
+				},
+				{
+					$lookup: {
+						from: 'rocketchat_livechat_department',
+						localField: '_id',
+						foreignField: '_id',
+						as: 'department',
+					},
+				},
+				{
+					$group: {
+						_id: {
+							$arrayElemAt: ['$department.name', 0],
+						},
+						total: {
+							$sum: '$total',
+						},
+					},
+				},
+				{
+					$sort: sort || { total: 1 },
+				},
+				{
+					$group: {
+						_id: null,
+						total: { $sum: '$total' },
+						data: {
+							$push: {
+								label: '$_id',
+								value: '$total',
+							},
+						},
+					},
+				},
+				{
+					$project: {
+						_id: 0,
+					},
+				},
+			],
+			{ readPreference: readSecondaryPreferred() },
+		);
+	}
+
+	getTotalConversationsWithoutDepartmentBetweenDates(start: Date, end: Date, extraQuery: Filter<IOmnichannelRoom>): Promise<number> {
+		return this.col.countDocuments({
+			t: 'l',
+			departmentId: {
+				$exists: false,
+			},
+			ts: {
+				$gte: start,
+				$lt: end,
+			},
+			...extraQuery,
+		});
+	}
+
+	getConversationsByTags(
+		start: Date,
+		end: Date,
+		sort: Record<string, 1 | -1>,
+		extraQuery: Filter<IOmnichannelRoom>,
+	): AggregationCursor<ReportResult> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						t: 'l',
+						ts: {
+							$lt: end,
+							$gte: start,
+						},
+						tags: {
+							$exists: true,
+							$ne: [],
+						},
+						...extraQuery,
+					},
+				},
+				{
+					$group: {
+						_id: '$tags',
+						total: {
+							$sum: 1,
+						},
+					},
+				},
+				{
+					$unwind: '$_id',
+				},
+				{
+					$group: {
+						_id: '$_id',
+						total: { $sum: '$total' },
+					},
+				},
+				{
+					$sort: sort || { total: 1 },
+				},
+				{
+					$group: {
+						_id: null,
+						total: { $sum: '$total' },
+						data: {
+							$push: {
+								label: '$_id',
+								value: '$total',
+							},
+						},
+					},
+				},
+				{
+					$project: {
+						_id: 0,
+					},
+				},
+			],
+			{ readPreference: readSecondaryPreferred() },
+		);
+	}
+
+	getConversationsWithoutTagsBetweenDate(start: Date, end: Date, extraQuery: Filter<IOmnichannelRoom>): Promise<number> {
+		return this.col.countDocuments({
+			t: 'l',
+			ts: {
+				$gte: start,
+				$lt: end,
+			},
+			$or: [
+				{
+					tags: {
+						$exists: false,
+					},
+				},
+				{
+					tags: {
+						$eq: [],
+					},
+				},
+			],
+			...extraQuery,
+		});
+	}
+
+	getConversationsByAgents(
+		start: Date,
+		end: Date,
+		sort: Record<string, 1 | -1>,
+		extraQuery: Filter<IOmnichannelRoom>,
+	): AggregationCursor<ReportResult> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						t: 'l',
+						ts: {
+							$gte: start,
+							$lt: end,
+						},
+						servedBy: {
```

Comment:
I'm not sure if it's possible for `servedBy` to be null also, but maybe it would be good if we check that or filter by `servedBy._id` instead. Wdyt?

Reply:
no code way of having servedBy as null, I'll check which one works better with the indexes